db/system_keyspace.cc

```diff
@@ -241,6 +241,7 @@ schema_ptr system_keyspace::topology() {
             .with_column("session", uuid_type, column_kind::static_column)
             .with_column("tablet_balancing_enabled", boolean_type, column_kind::static_column)
             .with_column("upgrade_state", utf8_type, column_kind::static_column)
+            .with_column("coordinator", uuid_type, column_kind::static_column)
```

Comment:
Feels wrong to store transient information in persistent state.

Better to make it available via an API call.

Reply:
Coordinator runs on the leader. The leader suppose to be available from "has to be added" raft state virtual table. It may bot be 100% reliable since different nodes may see different node as a leader during transition period, but the patch here is not reliable in the same situation.

Reply:
I see why adding state for transient things may feel wrong, but you must admin that the patch is pretty simple. 

Can we change this once https://github.com/scylladb/scylladb/issues/12008 is implemented?

Reply:
The patch piggybacks  on an existing write, which is nice.

Reply:
> I see why adding state for transient things may feel wrong, but you must admin that the patch is pretty simple.

I admin it.

> 
> Can we change this once #12008 is implemented?

If we do, then #12008 becomes a release blocker.

Don't all nodes have an idea who the leader is? For testing it should be sufficient to have an API to access that.

Reply:
https://github.com/scylladb/scylladb/issues/12008 looks like two days of work, we can make it release blocker if that would simplify everyone's life

On the other hand, we basically already made "current coordinator" persistent state through `fence_previous_coordinator()` function, and this is just extending the command already sent with more info. I like it.

But @tgrabiec the large comment at `fence_previous_coordinator()` needs to be updated, it says "Write empty change"

Reply:
What about this new state?

Reply:
@avikivity https://github.com/scylladb/scylladb/issues/12008 is not fixed yet. Both Gleb and Kamil seem to like this patch. Are you sure it's unacceptable?

The information which node is the coordinator may seem transient, but it's actually information which needs to be globally linearized with group0 operations, that's why we already have a group0 write on failover (in topology_coordinator::fence_previous_coordinator()). Making it part of group0 state makes it more explicit.

Reply:
Actually, I can't use this solution because it's prone to a race. The test assumes that after executing read barrier the info about the leader is accurate. But it's not guaranteed if we look at topology coordinator field, since it's updated with a delay. We need to check with the raft group directly.